gitpod-io/gitpod
# Add WS image metrics to workspace instances

## Description

Adds workspace instance metrics to track image sizes. This includes storing metrics in the database, exposing them through the workspace instance status, and making them available via the API's `WorkspaceService.ListWorkspaceSessions` endpoint.

## Related Issue(s)

Fixes CLC-1016

## How to test

https://ft-add-ws-3faf28c13f.preview.gitpod-dev.com/workspaces

Try this in your browser console after starting a workspace (fill in your org ID or [join mine](https://ft-add-ws-3faf28c13f.preview.gitpod-dev.com/orgs/join?inviteId=90fe4e89-8c65-4246-8b63-c208c5742188) so you don't have to):

```ts
const response = await fetch("https://ft-add-ws-3faf28c13f.preview.gitpod-dev.com/public-api/gitpod.v1.WorkspaceService/ListWorkspaceSessions", {
    "credentials": "include",
    "headers": {
        "User-Agent": "connect-es/1.1.2",
        "Accept": "*/*",
        "content-type": "application/json",
        "Sec-Fetch-Dest": "empty",
        "Sec-Fetch-Mode": "cors",
        "Sec-Fetch-Site": "same-origin",
        "Sec-GPC": "1",
    },
    "referrer": "https://ft-add-ws-3faf28c13f.preview.gitpod-dev.com/workspaces",
    "body": "{\"pagination\":{\"pageSize\":50},\"organizationId\":\"03a65446-6229-404d-b3ea-816b96533d85\"}",
    "method": "POST",
    "mode": "cors"
});
console.log(await response.json())
```

Under a workspace session in the response, you should see a `metrics` field.
